### PR TITLE
[WIP] add return_success_index option in csv file parser

### PR DESCRIPTION
**Pull request by corochann**
This PR adds return_success_index option.

We want to get which index preprocessing is succeed or failed in some situation 
- to investigate which smiles preprocessing is failed etc

At first, I want to get review that this design is ok or not.
After getting ok, I would like to proceed
 - SDFFileParser implementation
 - add test

### CODE REVIEW

**Comment by corochann** [comment]
Can you give some comments that it is ok to proceed with this direction or not?

File: chainer_chemistry/dataset/parsers/csv_file_parser.py [hunk 1]
```diff
@@ -161,15 +172,23 @@ def parse(self, filepath, return_smiles=False, target_index=None):
             result = pp.process(filepath)
```
[the remark this answers is not in the document]
Comment (mottodora) [reply chainer_chemistry/dataset/parsers/csv_file_parser.py]: We can't solve this problem in this PR.  I feel it is kind to raise error when `pp` is not an instance of `MolPreprocessor`. But it is not scope of this PR.

**Comment by mottodora** [comment]
Basic desing is OK. Can you proceed to implement?

**Comment by corochann** [comment]
Since I implemented in DataFrameParser, let me close this PR and re-submit in other PR.

**Pull request closed without merging**